src/Core/Hook/Provider/IdentifiableObjectHookByFormTypeProvider.php

```diff
@@ -0,0 +1,145 @@
+    const FORM_HANDLER_CREATE_AFTER_STARTS_WITH = 'actionAfterCreate';
+    const FORM_HANDLER_ENDS_WITH = 'FormHandler';
+
+    /**
+     * @var FormRegistryInterface
+     */
+    private $formFactory;
+
+    /**
+     * @param FormFactoryInterface $formFactory
+     */
+    public function __construct(FormFactoryInterface $formFactory)
+    {
+        $this->formFactory = $formFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getHookNames(array $formTypes)
+    {
+        $formNames = $this->getFormNames($formTypes);
+
+        $formBuilderHookNames = [];
+        $formHandlerBeforeUpdateHookNames = [];
+        $formHandlerAfterUpdateHookNames = [];
+        $formHandlerBeforeCreateHookNames = [];
+        $formHandlerAfterCreateHookNames = [];
+
+        foreach ($formNames as $formName) {
+            $formBuilderHookNames[] = $this->formatHookName(
+                self::FORM_BUILDER_HOOK_STARTS_WITH,
+                $formName,
+                self::FORM_BUILDER_HOOK_ENDS_WITH
+            );
+
+            $formHandlerBeforeUpdateHookNames[] = $this->formatHookName(
+                self::FORM_HANDLER_UPDATE_BEFORE_STARTS_WITH,
+                $formName,
+                self::FORM_HANDLER_ENDS_WITH
+            );
+
+            $formHandlerAfterUpdateHookNames[] = $this->formatHookName(
+                self::FORM_HANDLER_UPDATE_AFTER_STARTS_WITH,
+                $formName,
+                self::FORM_HANDLER_ENDS_WITH
+            );
+
+            $formHandlerBeforeCreateHookNames[] = $this->formatHookName(
+                self::FORM_HANDLER_CREATE_BEFORE_STARTS_WITH,
+                $formName,
+                self::FORM_HANDLER_ENDS_WITH
+            );
+
+            $formHandlerAfterCreateHookNames[] = $this->formatHookName(
+                self::FORM_HANDLER_CREATE_AFTER_STARTS_WITH,
+                $formName,
+                self::FORM_HANDLER_ENDS_WITH
+            );
+        }
+
+        return array_merge(
+            $formBuilderHookNames,
+            $formHandlerBeforeUpdateHookNames,
+            $formHandlerAfterUpdateHookNames,
+            $formHandlerBeforeCreateHookNames,
+            $formHandlerAfterCreateHookNames
+        );
+    }
+
+    /**
+     * Gets form names which are used when generating hooks.
+     *
+     * @param Definition[] $formTypes
+     *
+     * @return Generator
+     */
+    private function getFormNames(array $formTypes)
+    {
+        foreach ($formTypes as $formType) {
+            yield $this->formFactory->createBuilder($formType)->getName();
+        }
+    }
+
+    /**
+     * Formats hook names.
+     *
+     * @param string $hookStartsWith
+     * @param string $hookId
+     * @param string $hookEndsWidth
+     *
+     * @return string
+     */
+    private function formatHookName($hookStartsWith, $hookId, $hookEndsWidth)
+    {
+        //todo: camelise this if such functionality will be introduced to 1.7.6
```

Comment:
What do you mean with this comment?

Reply:
its related with https://github.com/PrestaShop/PrestaShop/pull/13643 but since it is going to be merged I'll remove this todo and use Container::camelise() here as well :)